packages/@lwc/module-resolver/src/index.ts

```diff
@@ -6,154 +6,113 @@
+        const namespacedModuleDir = path.join(modulesDir, ns);
+        if (fs.lstatSync(namespacedModuleDir).isDirectory()) {
+            const modules = fs.readdirSync(namespacedModuleDir);
+            modules.forEach(moduleName => {
+                const moduleDir = path.join(namespacedModuleDir, moduleName);
+                const entry = getModuleEntry(moduleDir, moduleName);
+                if (entry) {
+                    const specifier = `${ns}/${moduleName}`;
+                    resolvedModules.push({ entry, specifier });
+                }
+            });
         }
-    }
-    return config;
-}
+    });
 
-export function resolveModulesInDir(absPath: string): { [name: string]: RegistryEntry } {
-    return glob
-        .sync<FlatEntry>(MODULE_ENTRY_PATTERN, {
-            cwd: absPath,
-            transform: entry =>
-                typeof entry === 'string' ? { path: entry } : { path: entry.path },
-        })
-        .reduce((mappings, { path: file }) => {
-            const ext = path.extname(file);
-            const fileName = path.basename(file, ext);
-            const rootDir = path.dirname(file);
-            const rootParts = rootDir.split('/'); // the glob library normalizes paths to forward slashes only - https://github.com/isaacs/node-glob#windows
-            const entry = path.join(absPath, file);
-
-            const dirModuleName = rootParts.pop();
-            const dirModuleNamespace = rootParts.pop();
-            if (dirModuleNamespace && dirModuleName === fileName) {
-                const registryNode = createRegistryEntry(
-                    entry,
-                    `${dirModuleNamespace}/${fileName}`,
-                    fileName,
-                    dirModuleNamespace.toLowerCase()
-                );
-                mappings[registryNode.moduleSpecifier] = registryNode;
-            }
-
-            return mappings;
-        }, {});
+    return resolvedModules;
 }
 
-function hasModuleBeenVisited(module, visited) {
-    return visited.has(module);
-}
+export function resolveModulesFromNpm(packageName: string): RegistryEntry[] {
+    let resolvedModules: RegistryEntry[] = [];
+    try {
+        const pkgJsonPath = require.resolve(`${packageName}/package.json`);
+        const packageDir = path.dirname(pkgJsonPath);
+        const lwcConfigFile = path.join(packageDir, LWC_CONFIG_FILE);
 
-function expandModuleDirectories({
-    moduleDirectories,
-    rootDir,
-    modulePaths,
-}: Partial<ModuleResolverConfig> = {}) {
-    if (modulePaths) {
-        return modulePaths;
-    }
-    if (!moduleDirectories && !rootDir) {
-        // paths() is spec'd to return null only for built-in node
-        // modules like 'http'. To be safe, return empty array in
-        // instead of null in this case.
-        return require.resolve.paths('.') || [];
+        if (fs.existsSync(lwcConfigFile)) {
+            resolvedModules = resolveModules({ rootDir: lwcConfigFile });
```

Comment:
@diervo Can you confirm that the expected behavior is that if you have both a `lwc.config.js` and a `package.json` the config file takes precedence over the `package.json` config?